app/Libraries/Opengraph/HasOpengraph.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries\Opengraph;
+
+interface HasOpengraph
+{
```

Comment:
I wonder if this should require something like `$opengraphClass` and use it for `set_opengraph`. Not sure if helps anything at all though. And this currently isn't enforcing anything. One can have HasOpengraph model without corresponding class and it'll only explode on runtime. That said, specifying the class name won't help either since it's not checked and can't specify inheritance either.

eh 🤷

Reply:
I got rid of it since the type check is more of "belongs to `App\Models`" than anything 🤷